bundles/org.openhab.core.io.net/src/main/java/org/openhab/core/io/net/http/HttpClientFactory.java

```diff
@@ -64,4 +66,35 @@ public interface HttpClientFactory {
      * @return the shared Jetty http client
      */
     HttpClient getCommonHttpClient();
+
+    /**
+     * Creates a new Jetty HTTP/2 client.
+     * The returned client is not started yet. You have to start it yourself before using.
+     * Don't forget to stop a started client again after its usage.
+     * The client lifecycle should be the same as for your service.
+     * DO NOT CREATE NEW CLIENTS FOR EACH REQUEST!
+     *
+     * @param consumerName for identifying the consumer in the Jetty thread pool.
+     *            Must be between 4 and 20 characters long and must contain only the following characters [a-zA-Z0-9-_]
+     * @return the Jetty HTTP/2 client
+     * @throws NullPointerException if {@code consumerName} is {@code null}
```

Comment:
```suggestion
```

I think it is better to remove all these `@throws NullPointerException` from the interface. I see it was already part of the existing methods on the interface. Because the interface is annotated with `@NonNullByDefault` the "contract" is violated when `null` is passed as  `consumerName` so then all bets are off. Also when code is properly annotated with `@NonNullByDefault` the compiler will yell at the programmer to stop passing `null` values.

Reply:
Ok